jx/ui/JXMLComponent.js

```diff
@@ -127,13 +127,17 @@ JXMLComponent.prototype.setAttr = function(delta) {
 JXMLComponent.prototype.applyAttr = function(delta) {
 	if (this.root) { // component has internal structure
 		// Allow component to mangle attributes
-		if (this.resolved.onAttr)
-			this.resolved.onAttr(delta, this.attr);
+		if (this.resolved.onDelta)
+			this.resolved.onDelta(delta, this.attr);
 
-		JXML.deepMerge(this.attr, delta);
+		delta = JXML.mergeDiff(this.attr, delta);
+
+		if (this.resolved.onAttr && delta)
+			this.resolved.onAttr(this.attr, delta);
```

Comment:
do you need to merge the mangled `delta` to `this.attr` again after this statement ?